### ISSUE TITLE
[FEA]: Allow to find the K nearest points to a polygon (using quadtree)

### ISSUE THREAD

**Issue by fpenarruApr** [rapidsai/cuspatial#1324]
### Is this a new feature, an improvement, or a change to existing functionality?

New Feature

### How would you describe the priority of this feature request

Critical (currently preventing usage)

### Please provide a clear description of problem you would like to solve.

Right now, 
cuspatial.quadtree_point_to_nearest_linestring
allows to find the nearest point. It will be useful to find the N closest points to a feature (point, polygon or linestring)
Something similar to this article with Postgis:

https://www.crunchydata.com/blog/a-deep-dive-into-postgis-nearest-neighbor-search

```
SELECT loc.*, t.name, t.geom
  FROM locations loc
  JOIN LATERAL (SELECT name, geom FROM geonames gn
      ORDER BY loc.geom <-> gn.geom LIMIT 10) AS t ON true;
```

**Comment by fpenarruApr** [comment]
Hi @jarmak-nv 

Thanks for your help.
I think your suggestions are not going to solve my problem. top_n_distances will have the minimum distances between different points and polygons, and this is not what I want.  Let me clarify my problem.
Suppose points representing hospitals, and polygons representing buildings. A typical Facility Problem will be to find the 3 or 4 closest hospitals to each building.
So, the solution can be a dataframe with the same number of rows that buildings, and some fields like

id_building, id_hospital1, dist_hospital1, id_hospital2, dist_hospital2,  id_hospital3, dist_hospital3

 Later, we can assign colors to each polygon according to distances or some coefficients and see which areas could be with a bad service.
I hope now my problem is more clear.
Thank you very much for your help! :-)

**Comment by harrism** [comment]
Thanks for the great example!

**Comment by jarmak-nv** [comment]
@fpenarruApr Yeah I see that now! Definitely will leave this feature request open since it's not something we directly support. 

We'll have to get creative here, I'll test a few things out this week and post some ideas 🌐
